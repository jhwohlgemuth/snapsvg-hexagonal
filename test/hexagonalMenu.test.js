'use strict';

jest.useFakeTimers();

const Snap = require('snapsvg');
const {mockHexagonalMenu} = require('./mocks');
const HexagonalMenu = require('../lib/hexagonalMenu');

Object.assign(HexagonalMenu.prototype, mockHexagonalMenu);

let items = [
    {title: 'CHOICE ONE'},
    {title: 'CHOICE TWO'},
    {title: 'CHOICE THREE'},
    {title: 'CHOICE FOUR'},
    {title: 'CHOICE FIVE'},
    {title: 'CHOICE SIX', subtitle: 'this button is special'}
];
let sideLength = 10;
let width = 100;
let options = {sideLength, width};

describe('Hexagonal Menu', function() {
    let menu;
    beforeEach(() => {
        window.Snap = Snap;
        menu = new HexagonalMenu(items, options);
    });
    afterEach(() => {
        window.Snap = undefined;
    });
    it('can be created in the hidden state', () => {
        expect(menu.isVisible()).toBeFalsy();
        menu.buttons.forEach((button, index) => {
            let {sideLength, x, y} = button;
            expect(sideLength).toEqual(sideLength);
            expect({x, y}).toMatchSnapshot();
            expect(button.index).toEqual(index);
        });
    });
    it('can set titles and subtitles', () => {
        menu.buttons.forEach(button => {
            let {title} = button;
            expect(title).toMatchSnapshot();
        });
        let buttonWithSubtitle = menu.buttons.filter(button => (button.subtitle !== undefined));
        expect(buttonWithSubtitle[0].subtitle).toEqual(items[items.length - 1].subtitle);
    });
    it('can toggle visibility', () => {
        expect(menu.isVisible()).toBeFalsy();
        menu.toggle();
        jest.runTimersToTime(1000 * 1000);
        expect(menu.isVisible()).toBeTruthy();
        menu.toggle();
        jest.runTimersToTime(1000 * 1000);
    });
    it('can be reset', function() {
        menu.toggle();
        jest.runTimersToTime(1000 * 100);
        menu.buttons.forEach(button => {
            expect(button.isActive).toEqual(undefined);
        });
        menu.reset();
        jest.runTimersToTime(100);
        menu.buttons.forEach(button => {
            expect(button.isActive).toEqual(false);
        });
    });
});
