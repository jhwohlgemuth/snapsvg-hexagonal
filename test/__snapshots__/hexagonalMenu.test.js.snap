// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hexagonal Menu can be created in the hidden state 1`] = `
Object {
  "x": -118.66025403784438,
  "y": 4.999999999999999,
}
`;

exports[`Hexagonal Menu can be created in the hidden state 2`] = `
Object {
  "x": -118.66025403784438,
  "y": 26,
}
`;

exports[`Hexagonal Menu can be created in the hidden state 3`] = `
Object {
  "x": -118.66025403784438,
  "y": 47,
}
`;

exports[`Hexagonal Menu can be created in the hidden state 4`] = `
Object {
  "x": -118.66025403784438,
  "y": 68,
}
`;

exports[`Hexagonal Menu can be created in the hidden state 5`] = `
Object {
  "x": -118.66025403784438,
  "y": 89,
}
`;

exports[`Hexagonal Menu can be created in the hidden state 6`] = `
Object {
  "x": -118.66025403784438,
  "y": 110,
}
`;

exports[`Hexagonal Menu can set titles and subtitles 1`] = `"CHOICE ONE"`;

exports[`Hexagonal Menu can set titles and subtitles 2`] = `"CHOICE TWO"`;

exports[`Hexagonal Menu can set titles and subtitles 3`] = `"CHOICE THREE"`;

exports[`Hexagonal Menu can set titles and subtitles 4`] = `"CHOICE FOUR"`;

exports[`Hexagonal Menu can set titles and subtitles 5`] = `"CHOICE FIVE"`;

exports[`Hexagonal Menu can set titles and subtitles 6`] = `"CHOICE SIX"`;
